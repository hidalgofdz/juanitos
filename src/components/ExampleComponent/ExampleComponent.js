import React from 'react';
import PropTypes from 'prop-types';
import './ExampleComponent.scss';

const ExampleComponent = ({ name, onButtonClick }) => (
  <div className="example-component">
    <span>{`Hello ${name}`}</span>
    <button className="example-component__button" onClick={onButtonClick}>
      Pru√©bame
    </button>
  </div>
);

ExampleComponent.propTypes = {
  name: PropTypes.string.isRequired,
  onButtonClick: PropTypes.func
};

ExampleComponent.defaultProps = {
  onButtonClick: null
};

export default ExampleComponent;
