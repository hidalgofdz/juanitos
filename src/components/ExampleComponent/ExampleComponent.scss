.example-component {
  background-color: black;
  color: white;
  width: 100px;
}