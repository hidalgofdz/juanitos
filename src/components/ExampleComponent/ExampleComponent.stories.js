import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';

import ExampleComponent from './ExampleComponent';

storiesOf('components/ExampleComponent', module).add(
  'with default settings',
  () => {
    const name = text('name', 'juanito');

    return <ExampleComponent name={name} />;
  }
);
