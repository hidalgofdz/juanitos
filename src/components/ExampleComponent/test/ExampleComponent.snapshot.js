import React from 'react';
import { shallow } from 'enzyme';
import ExampleComponent from '../ExampleComponent';

describe('ExampleComponent snapshot', () => {
  test('it renders correctly', () => {
    const wrapper = shallow(<ExampleComponent name="juanito" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('it renders correctly with onButtonClick prop', () => {
    const wrapper = shallow(
      <ExampleComponent name="juanito" onButtonClick={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
