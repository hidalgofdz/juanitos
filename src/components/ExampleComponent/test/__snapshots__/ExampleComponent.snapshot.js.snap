// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent snapshot it renders correctly 1`] = `
<div
  className="example-component"
>
  <span>
    Hello juanito
  </span>
  <button
    className="example-component__button"
    onClick={null}
  >
    Pruébame
  </button>
</div>
`;

exports[`ExampleComponent snapshot it renders correctly with onButtonClick prop 1`] = `
<div
  className="example-component"
>
  <span>
    Hello juanito
  </span>
  <button
    className="example-component__button"
    onClick={[Function]}
  >
    Pruébame
  </button>
</div>
`;
